aten/src/ATen/native/cpu/CopyKernel.cpp

```diff
@@ -23,41 +23,71 @@ static void copy_kernel(TensorIterator& iter, bool non_blocking) {
-            [=](scalar_t a) -> scalar_t { return a; },
-            [=](Vec256<scalar_t> a) -> Vec256<scalar_t> { return a; });
-      });
-    } else if (isComplexType(dtype)) {
-      if (iter.tensor(0).is_conj() == iter.tensor(1).is_conj()) {
-        AT_DISPATCH_COMPLEX_TYPES(dtype, "copy_kernel", [&] {
-            cpu_kernel_vec(
+      } else if (dtype == ScalarType::ComplexHalf) {
+        cpu_kernel(iter, [=](c10::complex<at::Half> a) -> c10::complex<at::Half> { return a; });
+      } else if (isQIntType(dtype)) {
+        AT_DISPATCH_QINT_TYPES(dtype, "copy_kernel", [&] {
+          cpu_kernel_vec(
               iter,
               [=](scalar_t a) -> scalar_t { return a; },
               [=](Vec256<scalar_t> a) -> Vec256<scalar_t> { return a; });
-          });
+        });
+      } else if (isComplexType(dtype)) {
+        if (iter.tensor(0).is_conj() == iter.tensor(1).is_conj()) {
+          AT_DISPATCH_COMPLEX_TYPES(dtype, "copy_kernel", [&] {
+              cpu_kernel_vec(
+                iter,
+                [=](scalar_t a) -> scalar_t { return a; },
+                [=](Vec256<scalar_t> a) -> Vec256<scalar_t> { return a; });
+            });
+        } else {
+          AT_DISPATCH_COMPLEX_TYPES(dtype, "conj_kernel", [&] {
+              cpu_kernel_vec(
+                iter,
+                [=](scalar_t a) -> scalar_t { return conj_impl(a); },
+                [=](Vec256<scalar_t> a) -> Vec256<scalar_t> { return a.conj(); });
+            });
+        }
       } else {
-        AT_DISPATCH_COMPLEX_TYPES(dtype, "conj_kernel", [&] {
-            cpu_kernel_vec(
-              iter,
-              [=](scalar_t a) -> scalar_t { return conj_impl(a); },
-              [=](Vec256<scalar_t> a) -> Vec256<scalar_t> { return a.conj(); });
-          });
+        AT_DISPATCH_ALL_TYPES_AND2(
+            ScalarType::Bool, ScalarType::BFloat16,dtype, "copy_kernel", [&] {
+              cpu_kernel_vec(
+                  iter,
+                  [=](scalar_t a) -> scalar_t { return a; },
+                  [=](Vec256<scalar_t> a) { return a; });
+            });
       }
     } else {
-      AT_DISPATCH_ALL_TYPES_AND2(
-          ScalarType::Bool, ScalarType::BFloat16,dtype, "copy_kernel", [&] {
-            cpu_kernel_vec(
+      if (dtype == ScalarType::Half) {
+      cpu_kernel(iter, [=](at::Half a) -> at::Half { return -a; });
+      } else if (isComplexType(dtype)) {
+        if (iter.tensor(0).is_conj() == iter.tensor(1).is_conj()) {
+          AT_DISPATCH_COMPLEX_TYPES(dtype, "copy_kernel", [&] {
+              cpu_kernel_vec(
                 iter,
-                [=](scalar_t a) -> scalar_t { return a; },
-                [=](Vec256<scalar_t> a) { return a; });
-          });
+                [=](scalar_t a) -> scalar_t { return -a; },
+                [=](Vec256<scalar_t> a) -> Vec256<scalar_t> { return a.neg(); });
+            });
+        } else {
+          AT_DISPATCH_COMPLEX_TYPES(dtype, "conj_kernel", [&] {
```

Comment:
It's worth saying that this case of the condition should ~never actually happen (unless someone does `neg_view` but this is not a real thing)